base/accumulate.jl

```diff
@@ -247,6 +261,15 @@ function accumulate(op, A; dims::Union{Nothing,Integer}=nothing, kw...)
     accumulate!(op, out, A; dims=dims, kw...)
 end
 
+function accumulate(op, xs::Tuple; init = _InitialValue())
+    rf = BottomRF(op)
+    ys, = foldl(xs; init = ((), init)) do (ys, acc), x
+        acc = rf(acc, x)
```

Comment:
Note: this uses the machinery I added in #33526

https://github.com/JuliaLang/julia/blob/3720edfc06ee8687b54ee7fbb908c9f03f4fc557/base/reduce.jl#L73-L78